docs/core-concepts/raw-queries.md

```diff
@@ -115,75 +124,125 @@ await sequelize.query(
-* If an object is passed, `$key` is bound to `object['key']`. Each key must begin with a non-numeric char. `$1` is not a valid key, even if `object['1']` exists.
-* In either case `$$` can be used to escape a literal `$` sign.
+Inconsequentially to which database you use, in Sequelize bind parameters are written following a postgres-like syntax. You can either:
 
-The array or object must contain all bound values or Sequelize will throw an exception. This applies even to cases in which the database may ignore the bound parameter.
+- Use numeric identifiers (e.g. `$1`, `$2`, etc…). Note that these identifiers start at 1, not 0. The `bind` option must be an array which contains a value for each identifier used in the query (`$1` is bound to the 1st element in the array (`bind[0]`), etc…).
+- Use alphanumeric identifiers (e.g. `$firstName`, `$status`, etc…). These identifiers follow common identifier rules (alphanumeric & underscore only, cannot start with a number). The `bind` option must be a plain object which includes each bind parameter (without the `$` prefix).
 
-The database may add further restrictions to this. Bind parameters cannot be SQL keywords, nor table or column names. They are also ignored in quoted text or data. In PostgreSQL it may also be needed to typecast them, if the type cannot be inferred from the context `$1::varchar`.
+The `bind` option must contain all bound values, or Sequelize will throw an error.
+
+:::info
+
+Bind Parameters can only be used for data values. Bind Parameters cannot be used to dynamically change the name of a table, a column, or other non-data values parts of the query.
+
+Your database may have further restrictions with bind parameters.
+
+:::
+
+Examples:
 
 ```js
 const { QueryTypes } = require('@sequelize/core');
 
 await sequelize.query(
-  'SELECT *, "text with literal $$1 and literal $$status" as t FROM projects WHERE status = $1',
+  'SELECT * FROM projects WHERE status = $1',
   {
     bind: ['active'],
-    type: QueryTypes.SELECT
-  }
+    type: QueryTypes.SELECT,
+  },
 );
 
 await sequelize.query(
-  'SELECT *, "text with literal $$1 and literal $$status" as t FROM projects WHERE status = $status',
+  'SELECT * FROM projects WHERE status = $status',
   {
     bind: { status: 'active' },
-    type: QueryTypes.SELECT
-  }
+    type: QueryTypes.SELECT,
+  },
 );
 ```
+
+Sequelize does not currently support a way to [specify the DataType of a bind parameter](https://github.com/sequelize/sequelize/issues/14410).  
+Until such a feature is implemented, you can cast your bind parameters if you need to change their DataType:
+
+```js
+const { QueryTypes } = require('@sequelize/core');
+
+await sequelize.query(
+  'SELECT * FROM projects WHERE id = CAST($1 AS int)',
+  {
+    bind: [5],
+    type: QueryTypes.SELECT,
+  },
+);
+
+// some dialects (such as postgres) support a terser cast syntax you can use if you prefer:
```

Comment:
Should we make this a regular text with SQL example?

Reply:
Done